awx/main/scheduler/task_manager_models.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2022 Ansible by Red Hat
+# All Rights Reserved.
+import logging
+
+from django.conf import settings
+
+from awx.main.models import (
+    Instance,
+    InstanceGroup,
+)
+
+logger = logging.getLogger('awx.main.scheduler')
+
+
+class TaskManagerInstance:
+    """A class representing minimal data the task manager needs to represent an Instance."""
+
+    def __init__(self, obj, node_type, capacity, hostname):
```

Comment:
I'd prefer kwargs instead of position args, and you're already kind of invoking it this way.